epochStart/metachain/systemSCs.go

```diff
@@ -780,7 +866,7 @@ func (s *systemSCProcessor) EpochConfirmed(epoch uint32) {
 	for _, maxNodesConfig := range s.maxNodesEnableConfig {
 		if epoch >= maxNodesConfig.EpochEnable {
 			// to cover also rollbacks, we always set the maxNodes and set the Enabled flag
-			s.flagChangeMaxNodesEnabled.Toggle(true)
+			s.flagChangeMaxNodesEnabled.Toggle(epoch == maxNodesConfig.EpochEnable)
 			s.maxNodes = maxNodesConfig.MaxNumNodes
```

Comment:
s.maxNodes will be set to maxNodesConfig.MaxNumNodes even if epoch > maxNodesConfig.EpochEnable but 
s.flagChangeMaxNodesEnabled will be set to true only in epoch =  maxNodesConfig.EpochEnable. It is ok? If not you can revert the line 869 and change the condition in line 867 to epoch == maxNodesConfig.EpochEnable

Reply:
It is ok. Only at activation epoch should the function be called and max nodes to be set